lms/djangoapps/bookmarks/models.py

```diff
@@ -0,0 +1,53 @@
+"""
+This File contains Model for Bookmarks.
+"""
+
+from django.contrib.auth.models import User
+from django.db import models
+
+from jsonfield.fields import JSONField
+
+from model_utils.models import TimeStampedModel
+from xmodule.modulestore.django import modulestore
+from xmodule_django.models import CourseKeyField, LocationKeyField
+
+
+class Bookmark(TimeStampedModel):
+    """
+    Bookmarks model.
+    """
+    user = models.ForeignKey(User, db_index=True)
+    course_key = CourseKeyField(max_length=255, db_index=True)
+    usage_key = LocationKeyField(max_length=255, db_index=True)
+    display_name = models.CharField(max_length=255, default="", help_text="Display name of XBlock")
+    path = JSONField()  # "JSON, breadcrumbs to the XBlock"
+
+    @classmethod
+    def create(cls, bookmarks_data, block=None):
+        """
+        Create a bookmark object.
+        """
+        if not block:
+            block = modulestore().get_item(bookmarks_data['usage_key'])
+
+        bookmarks_data['display_name'] = block.display_name
+        bookmarks_data['path'] = cls.get_path(block)
+
+        bookmark, __ = cls.objects.get_or_create(**bookmarks_data)
+        return bookmark
+
+    @staticmethod
+    def get_path(block):
+        """
+        Returns List of dicts containing {"usage_id": "", display_name:""} for the XBlocks
```

Comment:
Can you document what "block" is expected to be? I'm not clear if it is a bookmarked unit, or if it can be any xblock.